setup.py

```diff
@@ -1,4 +1,70 @@
-from setuptools import setup
+import os
 
-if __name__ == "__main__":
-    setup()
+from setuptools import find_packages, setup
+
+# There is a two step process here. We need to run setup.py to know that we need to install pybind and pyarrow
+# (setup_requires)
+# So the first time setup.py is invoked, it goes through the exception, finds the setup_requires, installs pybind and
+# pyarrow, then restarts and goes through the try statement successfully, and can build the cpp extensions properly.
+# try:
```

Comment:
Does this imply you need to do `pip install -e .` twice to get the full feature set?